src/main/java/frc/robot/UserInterface.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Team 4909, Bionics
+ * Billerica Memorial High School
+ *
+ * Copyright:
+ *   2021 Bionics
+ *
+ * License:
+ *   MIT: https://opensource.org/licenses/MIT
+ *   See the LICENSE file in the project's top-level directory for details.
+ */
+
+package frc.robot;
+
+import java.util.HashMap;
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj2.command.button.JoystickButton;
+import frc.peyton.Drivetrain;
+import frc.bionic.UserInterfaceElement;
+import frc.bionic.swerve.command.DriveWithJoystick;
+
+@SuppressWarnings("rawtypes")
+@SuppressWarnings("unchecked")
+public class UserInterface
+{
+  // Registry of objects that our user interface can operate on
+  private static HashMap<String, UserInterfaceElement> objectRegistry = new HashMap<String, UserInterfaceElement>();
+
+  /**
+   * Add an object to the registry
+   *
+   * @param id
+   *   The ID to use for access to the provided object
+   *
+   * @param obj
+   *   The object accessed via reference by the provided ID
+   */
+  public static void registerObject(String id, UserInterfaceElement obj)
+  {
+    objectRegistry.put(id, obj);
+  }
+
+  /**
+   * Create the default user interface. All required objects are expected to
+   * have been registered before this function is called.
+   */
+  public static void createDefaultUI()
+  {
+    createUIJoystick0();
+    createUIJoystick1();
+    createUIDashboard();
+  }
+
+  /**
+   * Create the user interface operated via Joystick 0
+   */
+  private static void createUIJoystick0()
+  {
+    Joystick                           joystick0 = new Joystick(0);
+    UserInterfaceElement<Drivetrain>   drivetrainElem = objectRegistry.get("Drivetrain");
+    Drivetrain                         drivetrain = drivetrainElem.get();
```

Comment:
will this throw if the drivetrain is not populated in the object registry / is that expected?

Reply:
Yes, and it should. It's a programmer error to use an object that hasn't been made available. I could put in an explicit assert or test, but it's really not necessary since the crash caused by the throw will provide the stack trace and it'll get fixed immediately.

Reply:
Yep, makes sense, I was just thinking from the perspective of if down the line, we may try to configure UI input not at startup (i.e. endgame commands or similar), and when such errors would get thrown and if it would be handled gracefully, but that's really more of a usage consideration to just remain aware of.